src/eth/handlers/nft.ts

```diff
@@ -162,7 +164,11 @@ export function handleTransfer(
       if (nftActiveOrder) {
         const order = orders.get(nftActiveOrder.id);
         if (order) {
-          clearNFTOrderProperties(nft!);
+          const isComingBackToOrderOwner = order.owner === nft.owner.address;
+          order.status = isComingBackToOrderOwner
+            ? OrderStatus.open
+            : OrderStatus.transferred;
+          setNFTOrderTransferred(nft); // do not cancel active order
```

Comment:
If I'm not wrong, we need to, if there's an active order:
1. Put the current order in the transferred status
2. Check if the receiver has an open order and is the owner, put that order as open and set is as the active one.
If I'm not wrong, we're doing only the first one, right?